src/nft/components/collection/Card.tsx

```diff
@@ -0,0 +1,383 @@
+import clsx from 'clsx'
+import Column from 'components/Column'
+import { Box } from 'nft/components/Box'
+import { Row } from 'nft/components/Flex'
+import { MinusIconLarge, PlusIconLarge } from 'nft/components/icons'
+import { body, subheadSmall } from 'nft/css/common.css'
+import { themeVars, vars } from 'nft/css/sprinkles.css'
+import { useIsMobile } from 'nft/hooks'
+import { GenieAsset } from 'nft/types'
+import {
+  createContext,
+  MouseEvent,
+  ReactNode,
+  useContext,
+  useLayoutEffect,
+  useMemo,
+  useReducer,
+  useRef,
+  useState,
+} from 'react'
+
+import * as styles from './Card.css'
+
+/* -------- ASSET CONTEXT -------- */
+export interface CardContextProps {
+  asset: GenieAsset
+  hovered: boolean
+  selected: boolean
+  href: string
+  setHref: (href: string) => void
+}
+
+const CardContext = createContext<CardContextProps>({
+  asset: {} as GenieAsset,
```

Comment:
`Partial<GenieAsset>` i think?

Reply:
i guess if it's empty like this shouldn't asset just be undefined?

Reply:
solved by performing a check that context is used inside of provider

Reply:
i'm just trying to avoid casting the type here